### PR TITLE
Ecdsa sd 2023 algorithms

### PR DESCRIPTION
**Pull request by PatStLouis**
This adds some rdfc vectors as well as a create/verify proof algorithm checks.

For sd-2023, adds the base cryptosuite algorithms.

Next is to add sd functions as part of this file.

### CODE REVIEW

**Review by BigBlueHat — APPROVED** [review]
Overall, I think this is great. Sorry for the review time lag...

File: tests/vectors.js [hunk 1]
```diff
@@ -7,6 +7,168 @@ test vectors. */
 /* eslint-disable max-len */
 /* eslint-disable quote-props */
 /* eslint-disable quotes */
+export const ecdsaSdVectors = {
+  baseProof: {
+    "@context": [
+      "https://www.w3.org/ns/credentials/v2",
+      {
+        "@vocab": "https://windsurf.grotto-networking.com/selective#"
```
[the remark this answers is not in the document]
Comment (PatStLouis) [reply tests/vectors.js:15]: yeah these are taken from the spec:
https://www.w3.org/TR/vc-di-ecdsa/#example-signed-base-document

File: tests/vectors.js [hunk 2]
```diff
@@ -7,6 +7,168 @@ test vectors. */
...
+      "proofValue": "u2V0BhVhAkWKMO8zpRmfcUMksHUMtZM7cJt8PmNLsljKTYhSi8gZ7wAWnK4BrOZkrH3dZvxKWlnxGG_0xlFXmU5sa5-j71VgjgCQCKnLOGbY_FuM-ASpSkkOxsIR2E8n7Ml2q1UQ6tEwzi5OGWEBtAwxySlwwASjXlYLoLwyjdsIRYUa05OQzE0P4skx1-QJKi8HtGcJHtJfOTn7RhWKC0nkXODvUAChvnKDVY02TWEClAeVhEaVRWS_5ZfU6MlgnX07DuKm9XBS-b3RKfciM1Eu7L2mvkMebfBWVUw9WkuqNL4Tz-MkN4lXmljRt8r8DWEBqzbDH1twSPRKHu3zD0KP7YWlx4al1GFotnFizVkQbz6htSNvQIc7IfxLu7AO2fuhJbagJbZKlm-Vf7aNbXuuaWEA78ePS_Gy4lnOlUa2CgworZmUYq-Etff-9QK90v4xOWzNWvaILdcmkJs5zwH9b86f5yxos_NRK-VIO015zVlWTWEB2bYe3iy_95zezUdGp66X77IQbHhKunDI1BF0trlkrIkOCqviH4S1U3Nz4n5WWW8qsc3zAEq1Spquojg2mevHtWEA5EwtdjPeXn07NztIOTZ6PtyWt6v0Gfp5oFRAbvvcCpDnptz6gZd56Z6TBkI0eMUs4RI2g2rup607wVFuIUdvgpgBYIOGCDmZ9TBxEtWeCI9oVmRt0eHRGAaoOXx08gxL2IQt_AVggVkUuBrlOaELGVQWJD4M_qW5bcKEHWGNbOrPA_qAOKKwCWCBD6o5lQOWjNGwaTjq7H2Cn1-NPbwXLeDedy2YyiqL9TQNYIJEdvfdRibsv05I3pv8e6S1aUuAuBpGQHLhrYj4QX0knBFggk0AeXgJ4e6m1XsV5-xFud0L_1mUjZ9Mffhg5aZGTyDkFWCDYgT4e07o_IdCwae6qE7WZfpXtGRFESEXR3SxZmXE05o4AAQIFBggJCg4PEBESEw"
+    }
+  }
+};
+export const ecdsaRdfcVectors = {
+  'P-256': {
+    "@context": [
+      "https://www.w3.org/ns/credentials/v2",
+      "https://www.w3.org/ns/credentials/examples/v2"
```
[the remark this answers is not in the document]
Comment (PatStLouis) [reply tests/vectors.js:120]: I took the vectors directly from the spec, let me have a look.
https://www.w3.org/TR/vc-di-ecdsa/#example-signed-credential-0

File: tests/90-algorithms-sd.js [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      beforeEach(setupRow);
+      it('If the length of signatures does not match the length of ' +
+        'nonMandatory, an error MUST be raised and SHOULD convey an ' +
+        'error type of PROOF_VERIFICATION_ERROR, indicating that the ' +
+        'signature count does not match the non-mandatory message count.',
+      async function() {
+        this.test.link = 'https://www.w3.org/TR/vc-di-ecdsa/#proof-serialization-ecdsa-sd-2023';
+
+        // const validBaseProof = structuredClone(ecdsaSdVectors.baseProof);
+        // await verifySuccess(verifier, validBaseProof);
```
[the remark this answers is not in the document]
Comment (PatStLouis) [reply tests/90-algorithms-sd.js]: comment removed

**Pull request merged**